src/dual.jl

```diff
@@ -552,43 +552,73 @@ end
 # exponentiation #
 #----------------#
 
-for f in (:(Base.:^), :(NaNMath.pow))
```

Comment:
Could we avoid the code duplication and just change this to
```julia
for (f, invf) in ((:(Base.:^), :(Base.log)), (:(NaNMath.pow), :(NaNMath.log)))
```
and use `$invf` instead of `log` below?

Reply:
Yes